bff-service/server.js

```diff
@@ -0,0 +1,100 @@
+import http from 'http';
+import https from 'https';
+import { URL } from 'url';
+import NodeCache from 'node-cache';
+import 'dotenv/config';
+
+const PORT = process.env.PORT || 3000;
+const cache = new NodeCache({ stdTTL: 120 });
+
+// Mapping from env vars
+const serviceMap = {
+    'product': process.env.PRODUCT_SERVICE_URL,
+    'cart': process.env.CART_SERVICE_URL,
```

Comment:
with such an approach you'll need to **modify source code** and **redeploy** bff-service each time, when a new route is added.

Read again the requirements in Task 10.1:
> 3. Here's the workflow example that BFF Service should support:
> * Make requests to BFF Service with URL in the following format: `{bff-service-url}/{recipient-service-name}?var1=someValue`
> 
> * Get `recipientURL` from the env variables using `{recipient-service-name}` as a key.

the idea is to simplify maintenance. So when we want to add a new route `/new-svc`, we will just add a new environment variable like:
```
new-svc=https://new-url.com
```

and reload. no need to modify the source code!

Reply:
First of all, thank you for your comment.
But let me not agree with you.
1. According to the requirements,
`{recipient-service-name} - "cart" or "product" (you can use any other mapping of your choice)`
so the {recipient-service-name} from the piece of the requirements you mentioned is "cart" or "product" - or any other mapping of one's choice.
2. As for redeployment, there are advantages and disadvantages of each approach. If services are defined explicitly in the map, it's easier to see and maintain the list, we get more visibility and control. Using `process.env[varName]` directly may lead to unwanted drawbacks - e.g. when the varName is somehow the same as in node process - just console.log `Object.keys(process.env)` in the AWS environment and see there are much more keys than we define in .env file. Limiting keys ensures that the behavior is always the same - no matter of other env vars if the key is present on the map its the 1st flow, if it doesn't it's the 2nd flow - that's it.
Each of the approaches has advantages and drawbacks and the exact one should be selected based on the use-cases of the real system - which we don't have.
I can't agree with you that any of the approaches is a silver-bullet. :) 

Anyway, thank you very much for pointing this out!